crates/alkahest/src/util/action.rs

```diff
@@ -206,3 +242,120 @@ impl Action for SpawnRouteAction {
         false
     }
 }
+
+pub struct FollowAction {
+    route_ent: Entity,
+    traverse_from: Option<Entity>,
+}
+
+impl FollowAction {
+    pub fn new(route_ent: Entity, traverse_from: Option<Entity>) -> Self {
+        Self {
+            route_ent,
+            traverse_from,
+        }
+    }
+}
+
+impl Action for FollowAction {
+    fn start(&mut self, resources: &AppResources) {
+        let camera = resources.get::<Camera>();
+        let mut maps = resources.get_mut::<MapList>();
+
+        if let Some(map) = maps.current_map_mut() {
+            let scene = &mut map.scene;
+            let camera_offset = Vec3::Z;
+            let mut buffer = resources.get_mut::<ActionBuffer>();
+            const DEGREES_PER_SEC: f32 = 360.0;
+            const METERS_PER_SEC: f32 = 18.0;
+
+            let route_ref = scene.entity(self.route_ent);
+
+            let Some(route) = route_ref.get::<Route>() else {
+                return;
+            };
+
+            if let Some(children) = route_ref.get::<Children>() {
+                if children.0.is_empty() {
+                    return;
+                }
+
+                let start_index = match self.traverse_from {
+                    Some(e) => children.0.iter().position(|&ent| ent == e).unwrap_or(0),
+                    None => 0,
+                };
+
+                let start_node_ent = scene.entity(children.0[start_index]);
+                let start_pos = match start_node_ent.get::<Transform>() {
+                    Some(t) => t.translation,
+                    None => return,
+                };
+                let start_hash = match start_node_ent.get::<RouteNode>() {
+                    Some(n) => n.map_hash,
+                    None => return,
+                };
+
+                let mut old_pos = start_pos + camera_offset;
+                let mut old_orient = camera.get_look_angle(old_pos);
+                buffer.buffer_action(TweenAction::new(
+                    |x| x,
+                    Some((camera.position(), old_pos)),
+                    Some((camera.view_angle(), old_orient)),
+                    1.0,
+                ));
+
+                if let Some(hash) = start_hash {
+                    buffer.buffer_action(MapSwapAction::new(hash));
+                }
+                for node_e in children.0.iter().skip(start_index + 1) {
+                    let Some(pos) = scene.get::<Transform>(*node_e) else {
+                        return;
+                    };
+                    let Some(node) = scene.get::<RouteNode>(*node_e) else {
+                        return;
+                    };
+
+                    info!("Attempting to get next for {}", node_e);
```

Comment:
Don't think you want to clog the log with this

Reply:
woops. Debug log I forgot to remove.